### PR TITLE
Turn collector into its own package.

### PR DESCRIPTION
**Pull request by taknira**

### CODE REVIEW

**Comment by RJPercival** [comment]
Travis says you're missing a license header.

**Comment by taknira** [comment]
Fixed :)

File: collector/collector.go [hunk 1]
```diff
@@ -12,73 +12,81 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Collector runs a tool to collect the data needed to monitor a single
+// Package collector runs a tool to collect the data needed to monitor a single
```
Comment (RJPercival) [inline collector/collector.go]:
```suggestion
// Package collector fetches the data needed to monitor a single
```

File: collector/collector.go [hunk 2]
```diff
@@ -12,73 +12,81 @@
...
-	mmd            = flag.Duration("mmd", 24*time.Hour, "The Maximum Merge Delay for the Log")
-)
+// Config contains all of the configuration details for running the collector
+// for a particular Log.
+type Config struct {
+	// Details of the Log to collect data from.
+	Log *ctlog.Log
+	// How regularly the monitor should get an STH from the Log.
+	// To disable STH-getting, set to 0.
+	STHGetPeriod time.Duration
```
Comment (RJPercival) [inline collector/collector.go]: nit: Putting the "Get" at the start of the name makes them match the API names more closely, which makes the names a bit more memorable.
```suggestion
	GetSTHPeriod time.Duration
```

File: collector/collector.go [hunk 3]
```diff
@@ -12,73 +12,81 @@
...
+	storage.APICallWriter
+	storage.STHWriter
+}
 
-	ctx := context.Background()
-	l, err := ctlog.New(*logURL, *logName, *b64PubKey, *mmd)
-	if err != nil {
-		glog.Exitf("Unable to obtain Log metadata: %s", err)
+// Run runs the collector on the Log specified in cfg, and stores the collected
+// data in st.
```
Comment (RJPercival) [inline collector/collector.go]: Should mention that this doesn't return unless an error occurs or the context expires.

File: collector/datacollector/datacollector.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Datacollector runs a tool to collect the data needed to monitor Certificate
+// Transparency Logs, which can then be used to check that they are adhering to
+//RFC 6962.
```
Comment (RJPercival) [inline collector/datacollector/datacollector.go]:
```suggestion
// RFC 6962.
```

File: collector/datacollector/datacollector.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+		glog.Exitf("Unable to obtain Log metadata: %s", err)
+	}
+
+	cfg := &collector.Config{
+		Log:            l,
+		STHGetPeriod:   *sthGetPeriod,
+		RootsGetPeriod: *rootsGetPeriod,
+	}
+
+	collector.Run(ctx, cfg, &http.Client{}, &print.Storage{})
```
Comment (RJPercival) [inline collector/datacollector/datacollector.go]: Check for error returned by `collector.Run()`?

File: collector/collector.go (hunk 1 above)
Comment (taknira) [reply collector/collector.go]: Done.

File: collector/collector.go (hunk 2 above)
Comment (taknira) [reply collector/collector.go]: Done.

File: collector/collector.go (hunk 3 above)
Comment (taknira) [reply collector/collector.go]: Done.

File: collector/datacollector/datacollector.go (hunk 4 above)
Comment (taknira) [reply collector/datacollector/datacollector.go]: Done.

File: collector/datacollector/datacollector.go (hunk 5 above)
Comment (taknira) [reply collector/datacollector/datacollector.go]: Done.

**Pull request merged**